src/boot/CompressedAsyncStorage.js

```diff
@@ -0,0 +1,46 @@
+/* @flow */
+import { AsyncStorage, NativeModules } from 'react-native';
+
+export default class CompressedAsyncStorage {
+  // HACK: This wrapper around AsyncStorage.getItem() is not perfectly compliant with the
+  // specification of AsyncStorage.getItem(): `callback` should be optional, and getItem()
+  // should return the result. Here, we declare `callback` as non-optional, because redux-persist
+  // always calls getItem() with a callback. redux-persist is the only library that uses
+  // CompressedAsyncStorage. Also, we don't return the result, because redux-persist never uses it
+  // (instead, it looks at the result passed to `callback`). This avoids decompressing the result
+  // twice.
+  static async getItem(key: string, callback: (error: ?Error, result: ?string) => void) {
+    async function wrappedCallback(error, result) {
```

Comment:
I think this would actually be clearer by making this wrapper anonymous, like so:
```
await AsyncStorage.getItem(key, (error, result) => {
  // ... body of wrappedCallback ...
});
```

I was a bit confused at first reading this, until I saw below where this is passed to the underlying `getItem`. Swapping the order of presentation better matches the logical flow, I think.

(This is something that is awkward to do in Python, unfortunately, so we end up putting things in this inverted order a lot.)

Reply:
So we could do that, but then we couldn't apply the solution of your comment above anymore:
```js
const outerResult = await NativeModules.TextCompressionModule.decompress(result);
callback(error, outerResult);
return outerResult;
```

Reply:
I think we should favor matching the real `getItem()` in behavior over the readability aspect.

Reply:
Oh wait, actually, the solution in your comment above makes creating a wrapper function unnecessary, so we won't even have the readability problem :)